src/core/Form/FilterInput/FilterInput.tsx

```diff
@@ -0,0 +1,186 @@
+}
+
+class BaseFilterInput<T> extends Component<FilterInputProps<T>> {
+  render() {
+    const {
+      className,
+      inputContainerProps,
+      visualPlaceholder,
+      labelText,
+      labelMode,
+      optionalText,
+      status,
+      statusText,
+      id,
+      labelAlign,
+      'aria-describedby': ariaDescribedBy,
+      items: propItems,
+      onFiltering: propOnFiltering,
+      filterRule: propFilterRule,
+      ...passProps
+    } = this.props;
+
+    const onChangeHandler = (event: ChangeEvent<HTMLInputElement>) => {
+      const { items, onFiltering, filterRule } = this.props;
+      const { value } = event.target;
+
+      const filteredItems: T[] = items.reduce((filtered: T[], item: T) => {
+        if (filterRule(item, value)) {
+          filtered.push(item);
+        }
+        return filtered;
+      }, []);
+
+      onFiltering(filteredItems);
+    };
+
+    const statusTextId = statusText ? `${id}-statusText` : undefined;
+
+    return (
+      <HtmlDiv
+        {...inputContainerProps}
+        className={classnames(baseClassName, className, {
+          [filterInputClassNames.disabled]: !!passProps.disabled,
+          [filterInputClassNames.error]: status === 'error',
+        })}
+      >
+        <HtmlDiv
+          className={classnames(filterInputClassNames.wrapper, {
+            [filterInputClassNames.labelAlignLeft]: labelAlign === 'left',
+          })}
+        >
+          <LabelText
+            htmlFor={id}
+            as="label"
+            labelMode={labelMode}
+            optionalText={optionalText}
+          >
+            {labelText}
+          </LabelText>
+          <HtmlDiv className={filterInputClassNames.functionalityContainer}>
+            <HtmlDiv className={filterInputClassNames.inputElementContainer}>
+              <HtmlInput
+                {...passProps}
+                id={id}
+                className={filterInputClassNames.inputElement}
+                type="text"
+                placeholder={visualPlaceholder}
+                {...getConditionalAriaProp('aria-describedby', [
+                  statusTextId,
+                  ariaDescribedBy,
+                ])}
+                autoComplete="off"
+                aria-autocomplete="list"
+                auto-capitalize="false"
+                spellCheck="false"
+                onChange={onChangeHandler}
+              />
+            </HtmlDiv>
+            <StatusText id={statusTextId} status={status}>
+              {statusText}
+            </StatusText>
+          </HtmlDiv>
+        </HtmlDiv>
+      </HtmlDiv>
+    );
+  }
+}
+
+const FilterInputWithoutTokens: <T>(
+  props: FilterInputProps<T> & InternalTokensProp,
+) => JSX.Element = ({
+  // eslint-disable-next-line react/prop-types
+  tokens,
+  // eslint-disable-next-line react/prop-types
+  id: propId,
+  ...passProps
+}) => (
+  <AutoId id={propId}>
+    {(id) => <BaseFilterInput id={id} {...passProps} />}
+  </AutoId>
+);
```

Comment:
- `FilterInputWithoutTokens` is added to get types working with styled-components. Might be possible to combine it  with `StyledFilterInput` (down below) but I couldn't figure it out 😅 
- Better naming ideas are welcome.
- eslint disables are added as it warns from missing prop-types
  - did left them both to have own disable line, so that future additions changes are not affected (vs. using prop-type disable for whole file)